tests/area/test_event_dispatcher.py

```diff
@@ -0,0 +1,255 @@
+    return AreaDispatcher(area_with_markets)
+
+
+class TestAreaDispatcher:
+    """Collection of tests for AreaDispatcher."""
+
+    @staticmethod
+    def _get_agents_for_market_type(
+            dispatcher_object, market_type: AvailableMarketTypes
+    ) -> Dict[DateTime, Union[OneSidedAgent, BalancingAgent, SettlementAgent]]:
+        """Select the correct IAA dict in the AreaDispatcher depending on the market_type"""
+        if market_type == AvailableMarketTypes.SPOT:
+            return dispatcher_object.interarea_agents
+        if market_type == AvailableMarketTypes.BALANCING:
+            return dispatcher_object.balancing_agents
+        if market_type == AvailableMarketTypes.SETTLEMENT:
+            return dispatcher_object.settlement_agents
+        assert False, f"Market type not supported {market_type}"
+
+    @staticmethod
+    def _create_iaa_and_markets_for_time_slot(dispatcher_object: AreaDispatcher,
+                                              time_slot: DateTime,
+                                              market_class: Market,
+                                              market_type: AvailableMarketTypes):
+        """Helps to create iaas for testing."""
+        first_time_slot = time_slot
+        lower_market = MagicMock(autospec=market_class)
+        lower_market.time_slot = first_time_slot
+        higher_market = MagicMock(autospec=market_class)
+        higher_market.time_slot = first_time_slot
+
+        dispatcher_object.area.parent.get_market_instances_from_class_type = Mock(
+            return_value={first_time_slot: higher_market})
+
+        dispatcher_object.create_area_agents(market_type, lower_market)
+
+    # pylint: disable=too-many-arguments
+    @pytest.mark.parametrize("market_type, spot_market_type, market_class, expected_agent_type", [
+        [AvailableMarketTypes.SPOT, SpotMarketTypeEnum.ONE_SIDED.value, OneSidedMarket,
+         OneSidedAgent],
+        [AvailableMarketTypes.SPOT, SpotMarketTypeEnum.TWO_SIDED.value, TwoSidedMarket,
+         TwoSidedAgent],
+        [AvailableMarketTypes.SETTLEMENT, SpotMarketTypeEnum.TWO_SIDED.value, SettlementMarket,
+         SettlementAgent],
+        [AvailableMarketTypes.BALANCING, SpotMarketTypeEnum.TWO_SIDED.value, BalancingMarket,
+         BalancingAgent]
+    ])
+    def test_create_area_agents_creates_correct_objects(self, market_type: AvailableMarketTypes,
+                                                        spot_market_type: SpotMarketTypeEnum,
+                                                        market_class: Market,
+                                                        expected_agent_type: InterAreaAgent,
+                                                        area_dispatcher):
+        """Test if create_area_agents creates correct objects in the agent dicts."""
+        original_matching_type = ConstSettings.IAASettings.MARKET_TYPE
+        ConstSettings.IAASettings.MARKET_TYPE = spot_market_type
```

Comment:
Why is this needed, I couldn't find the reason behind it.
Reason for asking this is to check whether we can only pass either the `spot_market_type` or the `market_class`, not both

Reply:
`AreaDispatcher._create_agent_object` depends on this global setting